version/src/lib.rs

```diff
@@ -83,21 +76,40 @@ impl fmt::Debug for Version {
     fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
         write!(
             f,
-            "{}.{}.{} (src:{}; feat:{})",
+            "{}.{}.{} (src:{:08x}; feat:{}, client:{:?})",
             self.major,
             self.minor,
             self.patch,
-            match self.commit {
-                None => "devbuild".to_string(),
-                Some(commit) => format!("{commit:08x}"),
-            },
+            self.commit,
             self.feature_set,
+            self.client(),
         )
     }
 }
 
 impl Sanitize for Version {}
 
+impl From<u16> for ClientId {
+    fn from(client: u16) -> Self {
+        match client {
+            0u16 => Self::SolanaLabs,
+            _ => Self::Unknown(client),
+        }
+    }
+}
+
+impl TryFrom<ClientId> for u16 {
+    type Error = String;
+
+    fn try_from(client: ClientId) -> Result<Self, Self::Error> {
+        match client {
+            ClientId::SolanaLabs => Ok(0u16),
+            ClientId::Unknown(client @ 0u16..=0u16) => Err(format!("Invalid client: {client}")),
```

Comment:
What's this new rust syntax?

Reply:
`@ Bindings`?
https://doc.rust-lang.org/book/ch18-03-pattern-syntax.html#-bindings

Reply:
🎓 ty